### ISSUE TITLE
Whitespace after section header messes up with section namings

### ISSUE THREAD

**Issue by hroncok** [packit/specfile#417]
### What happened? What is the problem?

Hello, I was wondering why there isn't a build or install section found in [flowblade.spec](https://src.fedoraproject.org/rpms/flowblade/blob/50cab22e213a45b84a6f839057b436f6430e422f/f/flowblade.spec).

When I parse the spec, I get:

```pycon
>>> from specfile import Specfile
>>> specfile = Specfile('flowblade.spec')
>>> with specfile.sections() as sections:
...     for section in sections:
...         print(section.name, section.options)
... 
package 
description 
prep 
generate_buildrequires 
build 
install 
check 
files -f flowblade-trunk/%{name}.lang
changelog 
>>> with specfile.sections() as sections:
...     print("build" in sections)
...     print("install" in sections)
... 
False
False
```

I examined the specfile, and I found it has:

```
%build[space here]
...

%install[spec here]
```

Indeed, `sections.get("build ")` works.

I did not expect this to obfuscate the result of `"build" in sections` or `sections.build`. This means the only reliable way to get the build section is to iterate all the sections and look at the name.

### What did you expect to happen?

I expected the insignificant trailing whitespace to be ignored.

### Example URL(s)

https://src.fedoraproject.org/rpms/flowblade/blob/50cab22e213a45b84a6f839057b436f6430e422f/f/flowblade.spec

### Steps to reproduce

```bash
See above
```

### Workaround

- [X] There is an existing workaround that can be used until this issue is fixed.

### Participation

- [ ] I am willing to submit a pull request for this issue. (Packit team is happy to help!)

**Comment by nforro** [comment]
> This means the only reliable way to get the build section is to iterate all the sections and look at the name.

That's true, unfortunately. I think the original intention for this behavior was to be able to do e.g. `sections.get("package doc")`, but it makes more sense for the unique sections to be accessible without iterating through all.

**Comment by lbarcziova** [comment]
hi @hroncok. When discussing this within the team, we were not able to decide the priority of this issue, we would like to ask you whether this is a blocking issue for you/what is your use case for this functionality. Thanks in advance!

**Comment by lbarcziova** [comment]
Sure, we appreciate it. Thanks for the explanation, this helps us a lot with the followup planning decisions.
